### PR TITLE
Update docs for stable Hooks

### PR DESCRIPTION
**Pull request by bvaughn**
Fork of #1593 with release date modified.

### CODE REVIEW

**Comment by gaearon** [comment]
Also need Sunil’s change into the log. “Add TestUtils.act() to match browser behavior in tests” or something.

**Comment by gaearon** [comment]
(And changelog updates need to be reflected in React PR)

**Comment by bvaughn** [comment]
That's already in the blog post?

> We have added a new API called `ReactTestUtils.act()` in this release. It ensures that the behavior in your tests matches what happens in the browser more closely. We recommend to wrap any code rendering and triggering updates to your components into `act()` calls. Testing libraries like [`react-testing-library`](https://github.com/kentcdodds/react-testing-library) can also wrap their APIs with it.

File: content/blog/2019-02-05-react-v16.8.0.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  document.body.removeChild(container);
+  container = null;
+});
+
+it('can render and update a counter', () => {
+  // Test first render and componentDidMount
```
Comment (gaearon) [inline content/blog/2019-02-05-react-v16.8.0.md]: Oops this shouldn’t mention cDM

File: content/blog/2019-02-05-react-v16.8.0.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  act(() => {
+    ReactDOM.render(<Counter />, container);
+  });
+  const button = container.querySelector('button');
+  const label = container.querySelector('p');
+  expect(label.textContent).toBe('You clicked 0 times');
+  expect(document.title).toBe('You clicked 0 times');
+
+  // Test second render and componentDidUpdate
```
Comment (gaearon) [inline content/blog/2019-02-05-react-v16.8.0.md]: Same

File: content/docs/hooks-faq.md [hunk 3]
```diff
@@ -85,16 +105,78 @@ In the future, new versions of these libraries might also export custom Hooks su
...
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  document.body.removeChild(container);
+  container = null;
+});
+
+it('can render and update a counter', () => {
+  // Test first render and componentDidMount
```
Comment (gaearon) [inline content/docs/hooks-faq.md]: Same

File: content/docs/hooks-faq.md [hunk 4]
```diff
@@ -85,16 +105,78 @@ In the future, new versions of these libraries might also export custom Hooks su
...
+  act(() => {
+    ReactDOM.render(<Counter />, container);
+  });
+  const button = container.querySelector('button');
+  const label = container.querySelector('p');
+  expect(label.textContent).toBe('You clicked 0 times');
+  expect(document.title).toBe('You clicked 0 times');
+
+  // Test second render and componentDidUpdate
```
Comment (gaearon) [inline content/docs/hooks-faq.md]: Same

**Comment by gaearon** [comment]
>That's already in the blog post? 

I mean Changelog section in the end which should match the Changelog PR in React repo and lists all important changes.

File: content/blog/2019-02-05-react-v16.8.0.md (hunk 1 above)
Comment (gaearon) [reply content/blog/2019-02-05-react-v16.8.0.md]: Meaning, it should be “first render and effect”

**Comment by bvaughn** [comment]
Hm. I guess it seems really redundant since it's mentioned so explicitly right above, but sure I'll sync whatever Andrew adds to the CHANGELOG once he's added it.

File: content/blog/2019-02-05-react-v16.8.0.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+Even while Hooks were in alpha, the React community created many interesting [examples](https://codesandbox.io/react-hooks) and [recipes](https://usehooks.com) using Hooks for animations, forms, subscriptions, integrating with other libraries, and so on. We're excited about Hooks because they make code reuse easier, helping you write your components in a simpler way and make great user experiences. We can't wait to see what you'll create next!
+
+## Testing Hooks
+
+We have added a new API called `ReactTestUtils.act()` in this release. It ensures that the behavior in your tests matches what happens in the browser more closely. We recommend to wrap any code rendering and triggering updates to your components into `act()` calls. Testing libraries like [`react-testing-library`](https://github.com/kentcdodds/react-testing-library) can also wrap their APIs with it.
```
Comment (alexkrolick) [inline content/blog/2019-02-05-react-v16.8.0.md]: We came up with this for react-testing-library: https://github.com/kentcdodds/react-testing-library/pull/274

Should we migrate to the new API?

**Comment by gaearon** [comment]
By that logic we could also remove Hooks PR from it since Hooks are also mentioned :-)

We generally leave blog changelog consistent with what’s on GH release and in Changelog.md in the repo. It’s more of a detailed look if you want to see which PR implemented what, and the discussion. Or if you got breakage and want to see the exact list of changes. Whereas body of blog post is a friendly summary.

File: content/blog/2019-02-05-react-v16.8.0.md (hunk 5 above)
Comment (gaearon) [reply content/blog/2019-02-05-react-v16.8.0.md]: Migrate? These APIs don’t seem like they have anything in common except both relate to testing.

**Comment by bvaughn** [comment]
That's fine. I wasn't arguing against it, just saying it seemed redundant. I already pushed the update, as you'll see above 😄

File: content/blog/2019-02-05-react-v16.8.0.md (hunk 5 above)
Comment (kentcdodds) [reply content/blog/2019-02-05-react-v16.8.0.md]: Don't worry, I plan to personally update react-testing-library to handle this new API as soon as possible. People should experience little-to-no friction when upgrading existing hooks tests or refactoring class components to use hooks.

**Comment by gaearon** [comment]
Yeah I just wanted to explain why. It’s intentionally redundant. :-)

I forgot this but we should probably also add TestRenderer.act() to the reference. Maybe just “Same as TestUtils.act() but for the test renderer” with an explanation link.

**Pull request merged**